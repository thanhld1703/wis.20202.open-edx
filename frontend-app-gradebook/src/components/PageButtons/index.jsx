import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '@edx/paragon';

export default function PageButtons({
  prevPage, nextPage, selectedTrack, selectedCohort, selectedAssignmentType,
  getPrevNextGrades, match,
}) {
  return (
    <div
      className="d-flex justify-content-center"
      style={{ paddingBottom: '20px' }}
    >
      <Button
        style={{ margin: '20px' }}
        variant="outline-primary"
        disabled={!prevPage}
        onClick={() => getPrevNextGrades(
          prevPage,
          match.params.courseId,
          selectedCohort,
          selectedTrack,
          selectedAssignmentType,
        )}
      >
        Previous Page
      </Button>
      <Button
        style={{ margin: '20px' }}
        variant="outline-primary"
        disabled={!nextPage}
        onClick={() => getPrevNextGrades(
          nextPage,
          match.params.courseId,
          selectedCohort,
          selectedTrack,
          selectedAssignmentType,
        )}
      >
        Next Page
      </Button>
    </div>
  );
}

PageButtons.defaultProps = {
  match: {
    params: {
      courseId: '',
    },
  },
  nextPage: '',
  prevPage: '',
  selectedCohort: null,
  selectedTrack: null,
  selectedAssignmentType: null,
};

PageButtons.propTypes = {
  getPrevNextGrades: PropTypes.func.isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      courseId: PropTypes.string,
    }),
  }),
  nextPage: PropTypes.string,
  prevPage: PropTypes.string,
  selectedAssignmentType: PropTypes.string,
  selectedCohort: PropTypes.shape({
    name: PropTypes.string,
  }),
  selectedTrack: PropTypes.shape({
    name: PropTypes.string,
  }),
};
