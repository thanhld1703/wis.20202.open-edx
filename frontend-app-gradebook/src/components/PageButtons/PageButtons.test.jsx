import renderer from 'react-test-renderer';
import PageButtons from '.';

const createInput = function createInput(prevPage, nextPage) {
  return {
    prevPage,
    nextPage,
    selectedTrack: 't',
    selectedCohort: 'c',
    getPrevNextGrades() {},
  };
};

describe('PageButtons', () => {
  const assertPageButtonsSnapshot = function assertPageButtonsSnapshot(input) {
    const pb = renderer.create(PageButtons(input));
    const tree = pb.toJSON();
    expect(tree).toMatchSnapshot();
  };

  it('prev null, next null', () => {
    assertPageButtonsSnapshot(createInput(null, null));
  });

  it('prev null, next not null', () => {
    assertPageButtonsSnapshot(createInput(null, 'np'));
  });

  it('prev not null, next null', () => {
    assertPageButtonsSnapshot(createInput('pp', null));
  });

  it('prev not null, next not null', () => {
    assertPageButtonsSnapshot(createInput('pp', 'np'));
  });
});
