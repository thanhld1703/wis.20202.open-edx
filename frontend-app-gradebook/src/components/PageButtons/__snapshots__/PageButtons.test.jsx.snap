// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageButtons prev not null, next not null 1`] = `
<div
  className="d-flex justify-content-center"
  style={
    Object {
      "paddingBottom": "20px",
    }
  }
>
  <button
    className="btn btn-outline-primary"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "margin": "20px",
      }
    }
    type="button"
  >
    Previous Page
  </button>
  <button
    className="btn btn-outline-primary"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "margin": "20px",
      }
    }
    type="button"
  >
    Next Page
  </button>
</div>
`;

exports[`PageButtons prev not null, next null 1`] = `
<div
  className="d-flex justify-content-center"
  style={
    Object {
      "paddingBottom": "20px",
    }
  }
>
  <button
    className="btn btn-outline-primary"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "margin": "20px",
      }
    }
    type="button"
  >
    Previous Page
  </button>
  <button
    className="btn btn-outline-primary"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "margin": "20px",
      }
    }
    type="button"
  >
    Next Page
  </button>
</div>
`;

exports[`PageButtons prev null, next not null 1`] = `
<div
  className="d-flex justify-content-center"
  style={
    Object {
      "paddingBottom": "20px",
    }
  }
>
  <button
    className="btn btn-outline-primary"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "margin": "20px",
      }
    }
    type="button"
  >
    Previous Page
  </button>
  <button
    className="btn btn-outline-primary"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "margin": "20px",
      }
    }
    type="button"
  >
    Next Page
  </button>
</div>
`;

exports[`PageButtons prev null, next null 1`] = `
<div
  className="d-flex justify-content-center"
  style={
    Object {
      "paddingBottom": "20px",
    }
  }
>
  <button
    className="btn btn-outline-primary"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "margin": "20px",
      }
    }
    type="button"
  >
    Previous Page
  </button>
  <button
    className="btn btn-outline-primary"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "margin": "20px",
      }
    }
    type="button"
  >
    Next Page
  </button>
</div>
`;
